### PR TITLE
initial contents

### PR DESCRIPTION
**Pull request by yonderblue**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+      - uses: golangci/golangci-lint-action@v6
+        with:
+          version: latest
+      - name: mod tidy check
+        run: .github/scripts/go-mod-tidy-check.sh
+  test:
+    runs-on: ${{ matrix.os }}
+    strategy:
+      matrix:
+        os: [ubuntu-latest, macOS-latest, windows-latest]
```
Comment (pedro) [inline .github/workflows/ci.yml:20]: in-depth!!!!!11

I guess we could save some bucks here? unless there are differences expected across runtimes of course

**Review by pedro — APPROVED** [review]
looks good to me!

nice notes where files are coming from and such :+1:

File: .github/workflows/ci.yml (hunk 1 above)
Comment (yonderblue) [reply .github/workflows/ci.yml:20]: was going with what I had put in https://github.com/graxinc/kmutex, which always seems nice to ensure there is nothing platform specific. Too expensive? :D

Comment (pedro) [reply .github/workflows/ci.yml:20]: ahh nice!

yeeea, thinking now I'm sure it's peanuts hehe carry on!

**Pull request merged**